tap_github/streams.py

```diff
@@ -329,24 +384,25 @@ def sync_endpoint(self,
                                     if child in stream_to_sync:
 
                                         parent_id = tuple(record.get(key) for key in STREAMS[child]().id_keys)
-
-                                        # Sync child stream, if it is selected or its nested child is selected.
-                                        self.get_child_records(client,
-                                                            catalog,
-                                                            child,
-                                                            parent_id,
-                                                            repo_path,
-                                                            state,
-                                                            start_date,
-                                                            record.get(self.replication_keys),
-                                                            stream_to_sync,
-                                                            selected_stream_ids,
-                                                            parent_record = record)
+                                        if STREAMS[child]().id_keys and not all(parent_id):
+                                            pass
+                                        else:
+                                            # Sync child stream, if it is selected or its nested child is selected.
+                                            self.get_child_records(client,
+                                                                catalog,
+                                                                child,
+                                                                parent_id,
+                                                                repo_path,
+                                                                state,
+                                                                start_date,
+                                                                record.get(self.replication_keys),
+                                                                stream_to_sync,
+                                                                selected_stream_ids,
+                                                                parent_record = record)
```

Comment:
@MrHarigo `id_keys` is the value we will use to call the child_url, but for some children we don't need to call an API endpoint, so that's why I put the check, because some child streams can have no id_keys by design. By default the `STREAMS[nested_child]().id_keys` value is `[]`. Are you suggesting something like this:
```
if id_keys := STREAMS[nested_child]().id_keys:
   child_id = tuple(record.get(key) for key in id_keys)
   if not all(child_id): continue
self.get_child_records()
```

Reply:
@alanken10 if the `get_child_records()` works fine with the `parent_id` being equal to `[]` then the current code is fine 👍